src/Operation/Recommendation/CategoryMerchandising.php

```diff
@@ -57,12 +61,14 @@ public function __construct(
         AccountInterface $account,
         $customerId,
         $category,
+        Filters $filters,
```

Comment:
This should somehow indicate that we're talking about include filters. Maybe create `AbstractFilter` class and extend `IncludeFilters` and `ExcludeFilters` from that. This method constructor should take both `IncludeFilters` and `ExcludeFilters`.